assets/src/edit-story/components/inspector/inspectorProvider.js

```diff
@@ -52,6 +57,20 @@ function InspectorProvider({ children }) {
     setInspectorContentHeight(height)
   );
 
+  useEffect(() => {
+    if (selectedElementIds.length > 0 && tab === DOCUMENT) {
+      setTab(DESIGN);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedElementIds]);
+
+  useEffect(() => {
+    if (tab === DOCUMENT) {
+      setTab(DESIGN);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentPage]);
```

Comment:
I don't know if I'd rather have a single effect or two different. I guess this duplication makes sense, as it's all about dependency triggering.

Reply:
It's probably more clear this way since in the case of `currentPage` the change should happen even if no elements are selected.